config/reader.go

```diff
@@ -29,6 +32,14 @@ var (
 	GitRevision = ""
 )
 
+var (
+	placeholderRegexp       = regexp.MustCompile(`\$\{[A-Za-z0-9_\.\-]+\}`)
+	verifyPlaceholderRegexp = regexp.MustCompile(`^\$\{([A-Za-z0-9_\.\-]+)\}$`)
```

Comment:
So right now we don't have types for packages so ${packages.my_package} is a valid package (and is a ml-model subtype). There is a migration to add types to packages, but not sure which work will be done before so we need to support not having types here. 

I can definitely make this stricter to avoid other placeholders, but I wanted to make this so it would match on any placeholder that exists in the config. In theory its the way we match it paths that I think should change not the strictness of the this regex. In theory, we want to match and change ${HOME} if it exists in the config.

Reply:
So i updated the placeholderRegexp to be: `\$\{(packages[\.(ml_model|modules)]*\.[A-Za-z0-9_\-]+)}` -- this will allow for an optional type parameter of ml_model/modules to be passed in for now (which we can enforce later once this migration has happened). 

I think that if we want to start removing other placeholders, we should just add that functionality later when need be. Thanks for the suggestion, removed the verifyPlaceholder function for now as well

Reply:
New comment inline on the changed regex, so resolving this older thread.